macros/edr/data_monitoring/data_monitors_configuration/get_buckets_configuration.sql

```diff
@@ -47,8 +47,8 @@
 
     {%- set incremental_bucket_times_query %}
         with bucket_times as (
-            select min(last_bucket_end) as last_max_bucket_end,
-                   min(first_bucket_end) as last_min_bucket_end,
+            select max(last_bucket_end) as max_existing_bucket_end,
+                   {{ elementary.edr_cast_as_timestamp(elementary.edr_timeadd(metric_properties.time_bucket.period, -1 * metric_properties.time_bucket.count, 'min(first_bucket_end)')) }} as min_existing_bucket_start,
```

Comment:
Fixed here and below some inaccurate comparisons (e.g. comparing bucket starts to bucket ends)